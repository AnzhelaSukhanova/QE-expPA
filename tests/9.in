(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and 
 (or (bvult (bvmul x #b0100) (bvmul y #b0101)) (= (bvmul x #b0100) (bvmul y #b0101))) 
 (or (bvult (bvmul y #b1000) (bvmul x #b1010)) (= (bvmul y #b1000) (bvmul x #b1010))))))
