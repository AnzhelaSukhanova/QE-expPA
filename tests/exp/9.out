(let (($e1 (bvadd y (bvshl #b0001 y)))) (let (($e2 (bvadd #b0001 $e1))) (let (($e3 (ite (bvult $e2 #b1000) (ite (bvult $e2 #b0100) (ite (bvult $e2 #b0010) (ite (= $e1 #b1111) #b0100 #b0000) #b0001) #b0010) #b0011))) (let (($e4 (bvadd #b0001 $e3))) (let (($e5 (bvadd $e3 (bvshl #b0001 $e3)))) (let (($e6 (bvshl #b0001 $e4))) (let (($e7 (= $e1 #b0101))) (let (($e8 (bvult #b0110 $e2))) (not (and (not (bvult #b0011 $e2)) (not (bvult #b0001 $e2)) (not (= $e1 #b0010)) (not (= $e1 #b0000)) (not $e8) (not $e7) (not (bvult #b1011 $e2)) (not (= $e1 #b1010)) (not (bvult #b0100 $e2)) (not (= $e1 #b0011)) (not (bvult #b0101 $e2)) (not (= $e1 #b0100)) (not $e8) (not $e7) (not (bvult (bvadd $e4 $e6) $e2)) (not (bvult $e5 $e2)) (not (bvult #b0111 $e2)) (not (= (bvnot $e3) (bvadd (bvnot $e1) $e6))) (not (= $e2 $e5)) (not (= $e1 #b0110))))))))))))
